bin/pkg/list_locked_dependencies.ml

```diff
@@ -0,0 +1,113 @@
+open! Import
+open Pkg_common
+module Package_universe = Dune_pkg.Package_universe
+module Lock_dir = Dune_pkg.Lock_dir
+module Opam_repo = Dune_pkg.Opam_repo
+module Package_version = Dune_pkg.Package_version
+module Opam_solver = Dune_pkg.Opam_solver
+
+let info =
+  let doc = "List the dependencies locked by a lockdir" in
+  let man = [ `S "DESCRIPTION"; `P "List the dependencies locked by a lockdir" ] in
+  Cmd.info "list-locked-dependencies" ~doc ~man
+;;
+
+let package_deps_in_lock_dir_pp package_universe package_name ~transitive =
+  let traverse, traverse_word =
+    if transitive then `Transitive, "Transitive" else `Immediate, "Immediate"
+  in
+  let opam_package =
+    Package_universe.opam_package_of_package package_universe package_name
+  in
+  let list_dependencies which =
+    Package_universe.opam_package_dependencies_of_package
+      package_universe
+      package_name
+      ~which
+      ~traverse
+  in
+  Pp.concat
+    ~sep:Pp.newline
```

Comment:
This should be `Pp.cut`. We should almost never use `Pp.newline` because it breaks that are already open.

Reply:
It looks like `Pp.cut` doesn't guarantee that the line will be broken though. My intention here is to print a nested list of titled lists of the form:
```
- Dependencies of local package a.dev
  - dep1.0.0.1
  - dep2.0.0.1
  - ...
- Dependencies of local package b.dev
  - dep1.0.0.1
  - ...
- ...
```
I can mostly achieve this with `Pp.enumerate`  but the title of the sublists (the `Dependencies of local package ...` part) isn't an element of the list so I need another way to print a line of text above the list.

What does it mean to "break boxes that are already open"?

Reply:
I've got something that I think works using `Pp.vbox` to ensure that all the `cut`s cause a break. Let me know what you think.

Reply:
>  It looks like Pp.cut doesn't guarantee that the line will be broken though

If you want to try guarantee that a line will be broken, make a vertical box and then every space will be interpreted as a newline. See the implementation of `Pp.enumerate` for an example.

> What does it mean to "break boxes that are already open"?

Whenever you do `Pp.{box,vbox,hbox,hovbox,..} foo` we allow whoever is rendering `foo` to choose how to interpret `Pp.space` and `Pp.cut`. That allows us our errors to be well wrapped no matter the size of the user's terminal or if they're displayed as LSP diagnostics, to be wrapped there appropriately as well.

Therefore whenever you output a newline, you're breaking the layout engine's assumptions on how much more its allowed to render on this line before wrapping. Essentially, making your `Pp.t` value uncomposable since it cannot be wrapped with boxes.